hostresolver/hostresolver.go

```diff
@@ -0,0 +1,82 @@
+package hostresolver
+
+import (
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/patrickmn/go-cache"
+)
+
+// HostResolver used for host resolver
+type HostResolver struct {
+	Enabled      bool
+	Cache        *cache.Cache
+	ResolvConfig string
+	ResolvDepth  int
+}
+
+// NewHostResolver init host resolver
+func NewHostResolver(enabled bool, resConfig string, resDepth int) *HostResolver {
+	return &HostResolver{
+		Enabled:      enabled,
+		ResolvConfig: resConfig,
+		ResolvDepth:  resDepth,
+		Cache:        cache.New(cache.DefaultExpiration, 5*time.Minute),
+	}
+}
+
+// NewDefaultHostResolver init default host resolver
+func NewDefaultHostResolver() *HostResolver {
```

Comment:
Can you delete this function please?
Refer [here](https://github.com/containous/traefik/pull/3403/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR1570) for more details.